test/test-tty.c

```diff
@@ -260,7 +260,7 @@ TEST_IMPL(tty_file) {
 }
 
 TEST_IMPL(tty_pty) {
-#if defined(__ANDROID__)         || \
+#if !defined(__ANDROID__)        && \
     defined(__APPLE__)           || \
```

Comment:
`&&` has higher precedence than `||` so this is parsed as `#if (!defined(__ANDROID__) && defined(__APPLE__)) || ...`.